### ISSUE TITLE
Unable to add a DNS A record with alias to public ip address azure resource

### ISSUE THREAD

**Issue by ankur-soni-pia** [Azure/azure-cli#26274]
## Describe the bug

**Command Name**
`az network dns record-set a create`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
Expect <class 'int'>, got 300 (<class 'str'>)
Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 663, in execute
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 697, in _run_job
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 333, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 121, in handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/network/custom.py", line 2522, in create_dns_record_set
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/aaz/_command.py", line 149, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\lib\site-packages\azure\cli\command_modules\network\aaz_compact\latest\network\dns\record_set\__cmds.py", line 34, in _handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/aaz/_command.py", line 153, in _handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/aaz/_command_ctx.py", line 32, in __init__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/aaz/_field_value.py", line 76, in __setitem__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/aaz/_field_type.py", line 48, in process_data
azure.cli.core.aaz.exceptions.AAZInvalidValueError: Expect <class 'int'>, got 300 (<class 'str'>)
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az network dns record-set a create --resource-group {} --zone-name {} --name {} --target-resource {} --ttl {}`

## Environment Summary
```
Windows-10-10.0.22621-SP0
Python 3.10.10
Installer: MSI

azure-cli 2.48.1

Extensions:
azure-devops 0.26.0
desktopvirtualization 0.2.0
maintenance 1.3.0

Dependencies:
msal 1.20.0
azure-mgmt-resource 22.0.0
```

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by navba-MSFT** [comment]
@ankur-soni-pia Thanks for reaching out to us and reporting this issue. We are looking into this issue and we will provide an update.

**Comment by navba-MSFT** [comment]
@ankur-soni-pia The above PR has been filed to address this. It is currently under review.

**Comment by alexftr** [comment]
Do we have any workaround solution for this? 
At this moment we unable to create automatically DNS records from Azure Pipelines. This is causing a lot of issues.

**Comment by necusjz** [comment]
> Do we have any workaround solution for this? At this moment we unable to create automatically DNS records from Azure Pipelines. This is causing a lot of issues.

Move `--ttl` from create command to update command, e.g.,
```
az network dns record-set a update -n {} -z {} -g {} --set TTL=300
```

**Comment by alexftr** [comment]
Thanks for solution @necusjz 
Unfortunately for CNAME it doesn't work. 
But for CNAME, for now I removed "--ttl=300" from command and it creates records with default TTL=3600, which is ok for now.

**Comment by necusjz** [comment]
> Thanks for solution @necusjz Unfortunately for CNAME it doesn't work. But for CNAME, for now I removed "--ttl=300" from command and it creates records with default TTL=3600, which is ok for now.

Exactly, CNAME currently has no commands for updating.

**Comment by navba-MSFT** [comment]
@ankur-soni-pia The fix will be released around 05/23 in AzCLI 2.49.0. Workarounds can be found at https://github.com/Azure/azure-cli/issues/26274#issuecomment-1527066849.
